### PR TITLE
refactor: remove double checks of memtable size

### PR DESCRIPTION
**Pull request by cjwcommuny**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

In the `freeze_mutable` function, the method `memtables.mutable.is_empty()` is checked twice - once in `VersionControl::freeze_mutable` and again in `MemtableVersion::freeze_mutable`. These check could be combined so the `unwrap` can be eliminated.

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Review by killme2008 — APPROVED** [review]
Good catch!

**Review by v0y4g3r — APPROVED** [review]
LGTM

File: src/mito2/src/region/version.rs [hunk 1]
```diff
@@ -80,14 +80,12 @@ impl VersionControl {
     /// Freezes the mutable memtable if it is not empty.
     pub(crate) fn freeze_mutable(&self) -> Result<()> {
         let version = self.current().version;
-        if version.memtables.mutable.is_empty() {
-            return Ok(());
-        }
+
         // Safety: Immutable memtable is None.
```
Comment (killme2008) [inline src/mito2/src/region/version.rs:84]:
```suggestion
```

**Comment by killme2008** [comment]
Thank you.

**Pull request merged**